src/mds/Server.cc

```diff
@@ -5794,18 +5794,22 @@ int Server::parse_quota_vxattr(string name, string value, quota_info_t *quota)
       }
     } else if (name == "quota.max_bytes") {
       /*
-       * The "quota.max_bytes" must be aligned to 4MB if greater than or
-       * equal to 4MB, otherwise must be aligned to 4KB.
+       * The "quota.max_bytes" must be aligned to 4MiB if greater than or
+       * equal to 4MiB, otherwise must be aligned to 4KiB.
        */
       string cast_err;
       int64_t q = strict_iec_cast<int64_t>(value, &cast_err);
-      if(!cast_err.empty() ||
-         (!IS_ALIGNED(q, CEPH_4M_BLOCK_SIZE) &&
-          (q < CEPH_4M_BLOCK_SIZE && !IS_ALIGNED(q, CEPH_4K_BLOCK_SIZE)))) {
-        dout(10) << __func__ << ":  failed to parse quota.max_bytes: "
+      if(!cast_err.empty()) {
+        dout(10) << __func__ << ": failed to parse quota.max_bytes: "
                  << cast_err << dendl;
         return -CEPHFS_EINVAL;
       }
+      if (q >= CEPH_4M_BLOCK_SIZE && !IS_ALIGNED(q, CEPH_4M_BLOCK_SIZE) ||
+          q < CEPH_4M_BLOCK_SIZE && !IS_ALIGNED(q, CEPH_4K_BLOCK_SIZE)) {
+        dout(10) << __func__ << ": size must align to 4KiB if less than 4MiB or must align to 4MiB"
+                 << dendl;
+        return -CEPHFS_EINVAL;
+      }
```

Comment:
I might be lacking some context here but I feel there isn't any need to handle case of `4MiB` specially because indeed it is aligned with `4KiB`. Just having `!IS_ALIGNED(q, CEPH_4K_BLOCK_SIZE)` should suffice

Reply:
No, for example `5MiB` is a incorrect size. And if the size >= 4MiB it muse align to 4MiB.

Reply:
makes sense